### PR TITLE
Web3 age proof page

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Add AgeProof page

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        // This will be near instant, as we're just waiting for the Chrome async store
+        return null;
+    }
+
+    if (!sessionPasscode) {
+        return <Navigate to={absoluteRoutes.login.path} state={{ to: -1 }} />;
+    }
+
+    if (more) {
+        return <Web3ProofRequest onSubmit={onSubmit} onReject={onReject} />;
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx:189]: The header is shown as `Signature Request` when showing more.

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+            challenge: string;
+            url: string;
+        };
+    };
+}
+
+function useAgeFromStatement(statement: AtomicStatementV2) {
+    if (
+        statement.type !== 'AttributeInRange' ||
+        statement.attributeTag !== 'dob' ||
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx:54]:
```suggestion
        statement.attributeTag !== AttributeKeyString.dob ||
```

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    const statements: CredentialStatements = useMemo(() => parse(rawStatements), [rawStatements]);
+    const statement = statements[0].statement[0];
+
+    const identities = useConfirmedIdentities();
+    const credentials = useAtomValue(credentialsAtomWithLoading);
+
+    const credential = useMemo(() => {
+        const credentialStatement = statements[0];
+        if (!isAccountCredentialStatement(credentialStatement)) {
+            throw new Error('Unexpected error type');
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx:93]:
```suggestion
            throw new Error('Unexpected credential statement type');
```

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        }
+
+        return credentials.value.find((cred) => {
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx:96]: Try to re-use `getViableAccountCredentialsForStatement` instead.

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    }, [identities.loading, credentials.loading]);
+
+    const identity = credential ? identities.value.find((id) => isIdentityOfCredential(id)(credential)) : undefined;
+
+    const canProve = Boolean(credential);
+
+    const network = useAtomValue(networkConfigurationAtom);
+    const client = useAtomValue(grpcClientAtom);
+    const addToast = useSetAtom(addToastAtom);
+    // TODO log this;
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx:115]: Log what?

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+            <Button
+                className="flex-center age-proof__verify-button m-20"
+                onClick={() => {
+                    setCreatingProof(true);
+                    handleSubmit()
+                        .then(withClose(onSubmit))
+                        .catch((e) => {
+                            setCreatingProof(false);
+                            logError(e);
+                            // TODO what to do
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx:213]: It depends on what can cause this to fail.

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/i18n/en.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+const t = {
+    moreDetails: 'More details',
+    verify: 'Verify',
+    ageMin: 'Verify that you are older than {{ age }}',
+    ageMax: 'Verify that you are younger than {{ age }}',
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/i18n/en.ts:5]: Fix the texts, e.g. Verify that you are {{ age }} or younger.

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.scss [hunk 8]
```diff
@@ -0,0 +1,80 @@
+@import url('https://fonts.googleapis.com/css2?family=IBM+Plex+Sans:wght@300;400;500;700&display=swap');
+
+$t-font-family-ibm: 'IBM Plex Sans', sans-serif;
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.scss:3]: This is already defined in the typography file from Guillermo.

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.scss [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        padding: 10px 15px;
+        width: calc(100% - 40px);
+        margin: 20px;
+        align-items: center;
+    }
+
+    &__verify-button {
+        background: #fffde4;
+        color: $color-off-black;
+
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.scss:34]: add font-weight: 400.

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.scss [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    &__request {
+        background: linear-gradient(94.88deg, #005a78 -3.22%, #48a2ae 87.51%);
+        display: flex;
+        flex-direction: column;
+        justify-content: space-between;
+        min-height: 100%;
+        font-family: $t-font-family-ibm;
+    }
+
+    &__description {
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.scss:15]: Add some left/right margins to this.

**Review by orhoj — CHANGES_REQUESTED** [review]

File: examples/two-step-transfer/index.html [hunk 11]
```diff
@@ -197,6 +197,17 @@
                 provider.on('accountDisconnected', (accountAddress) => (currentAccountAddress = undefined));
                 provider.on('accountChanged', (accountAddress) => (currentAccountAddress = accountAddress));
                 provider.on('chainChanged', (chain) => alert(chain));
+                document.getElementById('ageProof').addEventListener('click', () => {
+                    const statement = new concordiumSDK.IdStatementBuilder().addMaximumAge(18).getStatement();
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply examples/two-step-transfer/index.html:201]: Yes, I also changed it to be web3Id

File: packages/browser-wallet/src/popup/pages/Login/Login.tsx [hunk 12]
```diff
@@ -55,7 +55,7 @@ export default function Login() {
         <div className="onboarding-setup__intro-wrapper">
             <div className="onboarding-setup__intro-wrapper__logos">
                 <Logo className="onboarding-setup__intro-wrapper__logo" />
-                <ConcordiumLetters className="onboarding-setup__intro-wrapper__concordium-letters" />
+                <ConcordiumLetters className="age-request__concordium-letters" />
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/pages/Login/Login.tsx:58]: Crazy

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx (hunk 5 above)
Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx:115]: The potential error, added it.

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx (hunk 6 above)
Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/pages/AgeProofRequest/AgeProofRequest.tsx:213]: I just "improved the message" and left it at that.

File: packages/browser-wallet/src/popup/pages/AgeProofRequest/i18n/en.ts (hunk 7 above)
Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/pages/AgeProofRequest/i18n/en.ts:5]: Changed it to conform to the way we write in the main proofs.

**Review by orhoj — APPROVED** [review]

**Pull request merged**